### PR TITLE
fix DAG storage correctness, performance, and robustness issues

### PR DESCRIPTION
**Pull request by stalep**
- KahnDagSort: replace PriorityQueue with ArrayDeque in isCircular(), add visited set to prevent infinite loops, throw on cycles in sort() instead of silently accepting, return ArrayList instead of COW list
- Node.dependsOn(): convert recursive DFS to iterative BFS with visited set, preventing StackOverflowError and exponential blowup on diamonds
- Node.hashCode(): remove expensive List.copyOf(sources) allocation
- Value.dependsOn(): add visited set to prevent exponential re-visiting
- Work: remove @Immutable so retryCount mutations persist correctly
- WorkQueue: replace CopyOnWriteArrayList with ArrayList (already under lock), optimize getScore() to short-circuit and use pendingWork set
- Add unique constraints on node_edge and value_edge join tables
- Add tests for cycle detection and BFS edge cases
- Replace hashCode() lazy initialization hacks in NodeService and ValueService with explicit sources.size() calls, since hashCode() no longer accesses the sources collection
- Add test application.properties with drop-and-create schema strategy to prevent stale SQLite columns from breaking tests

### CODE REVIEW

File: h5m-core/src/main/java/io/hyperfoil/tools/h5m/entity/Node.java [hunk 1]
```diff
@@ -47,8 +47,9 @@ public static enum ScalarVariableMethod { First, All}
     @ManyToMany(cascade = { CascadeType.PERSIST, CascadeType.MERGE }, fetch = FetchType.LAZY )
     @JoinTable(
             name="node_edge",
-            joinColumns = @JoinColumn(name = "child_id"), // Custom join column referencing the Student entity
-            inverseJoinColumns = @JoinColumn(name = "parent_id")
+            joinColumns = @JoinColumn(name = "child_id"),
+            inverseJoinColumns = @JoinColumn(name = "parent_id"),
+            uniqueConstraints = @UniqueConstraint(columnNames = {"child_id", "parent_id", "idx"})
```
[the remark this answers is not in the document]
Comment (stalep) [reply h5m-core/src/main/java/io/hyperfoil/tools/h5m/entity/Node.java]: The current constraint (child_id, parent_id, idx) prevents the exact same edge at the same position, but allows the same parent to appear at different positions in the ordered list. Whether that's a
   bug depends on intent — if sources is an @OrderColumn list, Hibernate manages idx itself, and having (child_id, parent_id) as unique would prevent a node from depending on the same parent via two different input slots. If the domain allows that, the current constraint is fine. If not, Copilot's suggestion of splitting into (child_id, parent_id) + (child_id, idx) is better.
